geode-redis/src/distributedTest/java/org/apache/geode/redis/SaddDistDunitTest.java

```diff
@@ -0,0 +1,239 @@
+
+  @ClassRule
+  public static ClusterStartupRule clusterStartUp = new ClusterStartupRule(4);
+
+  static final String LOCAL_HOST = "127.0.0.1";
+  static final int SET_SIZE = 1000;
+  static Jedis jedis1;
+  static Jedis jedis2;
+  static Jedis jedis3;
+
+  static Properties locatorProperties;
+  static Properties serverProperties1;
+  static Properties serverProperties2;
+  static Properties serverProperties3;
+
+  static MemberVM locator;
+  static MemberVM server1;
+  static MemberVM server2;
+  static MemberVM server3;
+
+  @BeforeClass
+  public static void classSetup() {
+    locatorProperties = new Properties();
+    serverProperties1 = new Properties();
+    serverProperties2 = new Properties();
+    serverProperties3 = new Properties();
+
+    locatorProperties.setProperty(MAX_WAIT_TIME_RECONNECT, "15000");
+
+    serverProperties1.setProperty(REDIS_PORT, "6371");
+    serverProperties1.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    serverProperties2.setProperty(REDIS_PORT, "6372");
+    serverProperties2.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    serverProperties3.setProperty(REDIS_PORT, "6373");
+    serverProperties3.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    locator = clusterStartUp.startLocatorVM(0, locatorProperties);
+    server1 = clusterStartUp.startServerVM(1, serverProperties1, locator.getPort());
+    server2 = clusterStartUp.startServerVM(2, serverProperties2, locator.getPort());
+    server3 = clusterStartUp.startServerVM(3, serverProperties3, locator.getPort());
+
+    jedis1 = new Jedis(LOCAL_HOST, 6371);
+    jedis2 = new Jedis(LOCAL_HOST, 6372);
+    jedis3 = new Jedis(LOCAL_HOST, 6373);
+  }
+
+  @Before
+  public void testSetup() {
+    jedis1.flushAll();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    jedis1.disconnect();
+    jedis2.disconnect();
+    jedis3.disconnect();
+
+    server1.stop();
+    server2.stop();
+    server3.stop();
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenMultipleClients() {
+
+    String key = "key";
+
+    List<String> members = makeMemberList(SET_SIZE, "member1-");
+
+    jedis1.sadd(key, members.toArray(new String[] {}));
+
+    Set<String> result = jedis2.smembers(key);
+
+    assertThat(result.toArray()).containsExactlyInAnyOrder(members.toArray());
+
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenMultipleClients_AddingDifferentDataToSameSetConcurrently()
+      throws InterruptedException {
+
+    String key = "key";
+
+    List<String> members1 = makeMemberList(SET_SIZE, "member1-");
+    List<String> members2 = makeMemberList(SET_SIZE, "member2-");
+
+    Runnable addSetsWithClient1 = makeSADDRunnable(key, members1, jedis1);
+    Runnable addSetsWithClient2 = makeSADDRunnable(key, members2, jedis2);
+
+    runConcurrentThreads(addSetsWithClient1, addSetsWithClient2);
+
+    Set<String> results = jedis3.smembers(key);
+
+    members1.addAll(members2);
```

Comment:
I think this method would be clearer if at the top, when you initialize members1 and members2 you also created a list of "allMembers" by adding members1 and members2 to it. I think this is better than mutating members1 by adding members2 to it.

Reply:
I agree, that's clearer. Implemented!